packages/api/lambdas/publish-reports.js

```diff
@@ -0,0 +1,267 @@
+      granule,
+      eventMessage,
+      executionUrl,
+      executionDescription
+    );
+    return publishGranuleSnsMessage(granuleRecord);
+  } catch (err) {
+    log.error('Error handling granule from message', err);
+    log.info('Granule from message', granule);
+    log.info('Execution message', eventMessage);
+    return Promise.resolve();
+  }
+}
+
+/**
+ * Publish individual granule messages to SNS topic.
+ *
+ * @param {Object} eventMessage - Workflow execution message
+ * @returns {Promise}
+ */
+async function handleGranuleMessages(eventMessage) {
+  const granules = getMessageGranules(eventMessage);
+  if (!granules) {
+    log.info('No granules to process on the message');
+    return Promise.resolve();
+  }
+
+  const executionArn = getMessageExecutionArn(eventMessage);
+  const executionUrl = getExecutionUrl(executionArn);
+
+  let executionDescription;
+  try {
+    executionDescription = await StepFunctions.describeExecution({ executionArn });
+  } catch (err) {
+    log.error(`Could not describe execution ${executionArn}`, err);
+  }
+
+  try {
+    return Promise.all(
+      granules
+        .filter((granule) => granule.granuleId)
+        .map((granule) => buildAndPublishGranule(
+          granule,
+          eventMessage,
+          executionUrl,
+          executionDescription
+        ))
+    );
+  } catch (err) {
+    log.error('Error handling granule records', err);
+    log.info('Execution message', eventMessage);
+    return Promise.resolve();
+  }
+}
+
+/**
+ * Publish PDR record to SNS topic.
+ *
+ * @param {Object} eventMessage - Workflow execution message
+ * @returns {Promise}
+ */
+async function handlePdrMessage(eventMessage) {
+  try {
+    const pdrRecord = Pdr.generatePdrRecord(eventMessage);
+    if (!pdrRecord) return Promise.resolve();
+    return publishPdrSnsMessage(pdrRecord);
+  } catch (err) {
+    log.error('Error trying to generate PDR', err);
+    log.info('Execution message', eventMessage);
+    return Promise.resolve();
+  }
+}
+
+/**
+ * Publish messages to SNS report topics.
+ *
+ * @param {Object} eventMessage - Workflow execution message
+ * @param {boolean} isTerminalStatus - true if workflow is in a terminal state
+ * @param {boolean} isFailedStatus - true if workflow is in a failed state
+ * @returns {Promise}
+ */
+async function publishReportSnsMessages(eventMessage, isTerminalStatus, isFailedStatus) {
+  let status;
+
+  if (isTerminalStatus) {
+    status = isFailedStatus ? 'failed' : 'completed';
+  } else {
+    status = 'running';
+  }
+
+  merge(eventMessage, {
+    meta: {
+      status
+    }
+  });
+
+  return Promise.all([
+    handleExecutionMessage(eventMessage),
+    handleGranuleMessages(eventMessage),
+    handlePdrMessage(eventMessage)
+  ]);
+}
+
+/**
+ * Lambda handler for publish-reports Lambda.
+ *
+ * @param {Object} event - Cloudwatch event
+ * @returns {Promise}
+ */
+async function handler(event) {
```

Comment:
This handler is currently unused. It will be used once this new publish-reports Lambda is actually triggered by Cloudwatch events